pkg/diff/diff.go

```diff
@@ -0,0 +1,71 @@
+package diff
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-openapi/swag"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/treeverse/lakefs/pkg/api"
+	"github.com/treeverse/lakefs/pkg/local"
+)
+
+const diffTypeTwoDot = "two_dot"
+
+func GetDiffRefs(ctx context.Context, client api.ClientWithResponsesInterface, repository, left, right, prefix string, diffs chan<- api.Diff, twoDot bool) error {
+	defer func() {
+		close(diffs)
+	}()
+	var diffType *string
+	if twoDot {
+		diffType = api.StringPtr(diffTypeTwoDot)
+	}
+
+	// left should be the base commit!
+	hasMore := true
+	var after string
+	for hasMore {
+		diffResp, err := client.DiffRefsWithResponse(ctx, repository, left, right, &api.DiffRefsParams{
+			After:  (*api.PaginationAfter)(swag.String(after)),
+			Prefix: (*api.PaginationPrefix)(&prefix),
+			Type:   diffType,
+		})
+		if err != nil {
+			return err
+		}
+		if diffResp.HTTPResponse.StatusCode != http.StatusOK {
+			return fmt.Errorf("HTTP %d: %w", diffResp.StatusCode(), local.ErrRemoteDiffFailed)
+		}
+
+		for _, d := range diffResp.JSON200.Results {
+			diffs <- d
+		}
+
+		hasMore = diffResp.JSON200.Pagination.HasMore
+		after = diffResp.JSON200.Pagination.NextOffset
+	}
+	return nil
+}
+
+func Fmt(change string) (string, text.Color) {
+	var color text.Color
+	var action string
+
+	switch change {
+	case "added":
+		color = text.FgGreen
+		action = "+ added"
+	case "removed":
+		color = text.FgRed
+		action = "- removed"
+	case "changed", "modified":
+		color = text.FgYellow
+		action = "~ modified"
+	case "conflict":
+		color = text.FgHiYellow
+		action = "* conflict"
```

Comment:
Just noticed it's the color of conflict is a bit similar to the modified color, not sure we want to do anything about it, but I had to put it out there 😄

Reply:
Yes - this is how diff behaves this far and I maintained this also for the local status. I'm for changing the color, but since I feel there will be strong opinions about which colors should thing be - we should better deffer it to another issue